File: services/tradelog/tradelog-api/src/main/java/com/example/tradelog/api/converter/protobuf/ActionTypeConverter.java
package com.example.tradelog.api.converter.protobuf;

import java.util.function.Function;

public class ActionTypeConverter {
    public static Function<com.example.tradelog.api.spec.model.ActionType, com.example.tradelog.api.core.model.ActionType> toModel = protoModel -> {
        return com.example.tradelog.api.core.model.ActionType.Companion.lookup(protoModel.name());
    };

    public static Function<com.example.tradelog.api.core.model.ActionType, com.example.tradelog.api.spec.model.ActionType> toProto = apiModel -> {
        return com.example.tradelog.api.spec.model.ActionType.valueOf(apiModel.name());
    };
}

File: services/gateway/gateway-api/src/main/kotlin/com/example/gateway/api/rest/converter/ActionTypeConverter.kt
package com.example.gateway.api.rest.converter

import com.example.gateway.api.core.model.ActionType
import com.example.gateway.api.spec.model.GWShareJournalDto
import com.example.tradelog.api.spec.model.ShareJournalDto as TLShareJournalDto

class ActionTypeConverter {

    companion object {
        fun toModel(dto: TLShareJournalDto.ActionType): ActionType {
            return ActionType.lookup(dto.name)
        }

        fun toDto(model: ActionType): GWShareJournalDto.ActionType {
            return GWShareJournalDto.ActionType.valueOf(model.name)
        }
    }
}